### ISSUE TITLE
`gh-pages` CI job on PRs is erroring out

### ISSUE THREAD

**Issue by rgommers** [data-apis/array-api#466]
The preview CI job seems to have an issue right now, from [this log](https://github.com/data-apis/array-api/runs/7542519416?check_suite_focus=true) from gh-464:
```
Run git push "***github.com/data-apis/array-api.git"
remote: Permission to data-apis/array-api.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/data-apis/array-api.git/': The requested URL returned error: 403
Error: Process completed with exit code 128.
```
It seems to be misconfigured, so it works for PRs made from a branch within this repo, but fails when it's from a fork. We shouldn't use the former anyway, @kgryte can you please send PRs from your fork and clean up the many branches in this repo?

**Comment by jakirkham** [comment]
Maybe a token expired and needs to be regenerated/readded?

**Comment by rgommers** [comment]
Fixed now. There were two problems:
1. The publish workflow was running on branches other than `main`
2. The preview CI job on CircleCI was not triggering on forks.

To fix (2), I had to toggle this button:

<img width="909" alt="image" src="https://user-images.githubusercontent.com/98330/181728804-e4c1879c-5c5e-4934-b3a6-05ed43ad49fa.png">

**Comment by rgommers** [comment]
Third issue is that the permissions issue reported above is still happening on `main`.

**Comment by rgommers** [comment]
That last issue fixed in commit 792f11e864.
